notebooks/concepts/Mosviz concept notebook.ipynb

```diff
@@ -0,0 +1,542 @@
+   "outputs": [],
+   "source": [
+    "tab = mosapp.get_data('table')\n",
+    "tab['redshift']  #<- a regular Astropy table column but with whatever the current state of the table in the UI\n",
+    "tab.write('output.ecsv')  "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "* In the desktop app (but *not* the notebook, if possible) there can be a save button that does exactly the above cell."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Spectrum interactions\n",
+    "\n",
+    "* When the 1d spectrum is zoomed along the x axis, the 2d spectrum should be matched to it (and vice versa).  Y-axis of both should be independent.\n",
+    "\n",
+    "## Keyboard interactions\n",
+    "\n",
+    "* There need to be keyboard shortcuts ('{'/'}', or '<'/'>'?  Or tab/ctrl-tab?) that advance to the next spectrum.\n",
+    "* There needs to be a way to quickly update a column in a table. e.g. you press \"1\" to label it quickly as \"good spectrum\" or \"2\" as \"bad spectrum\". (this might be a plugin?)\n",
+    "* The user should be able to customize that \"quickly update column\".  As a suggestion for how this might work:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mosapp.set_keyboard_column_mappings({'1': ('spectrum_quality', 'good'), \n",
+    "                                     '2': ('spectrum_quality', 'bad'), })"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mosapp.set_keyboard_column_mappings('spectrum_quality')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Which tells mosviz that when the user presses the '1' key, that row's 'spectrum_quality' column should be updated with the value 'good', whereas '2' should update it to 'bad'.\n",
+    "\n",
+    "* This custom mapping might also be implemented as a plugin, but that should come *after* getting the notebook version working."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Additional notebook Glue selection interactions "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If the user wants the \"full power\" of Glue working with the table, a quick pathway is to have them create additional viewers to create subsets.  For example, suppose the `table` has two rows, one for 'rmag' and another for 'gmag', indicating the 'r' and 'g'-band magnitudes of the objects the spectra are of.  The user could then do:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tab = mosapp.get_data('table')\n",
+    "tab['color'] = tab['gmag'] - tab['rmag']  #<- this should update the data table...\n",
+    "\n",
+    "mosapp.scatter2d('color', 'rmag')"
```

Comment:
Minor: glue-jupyter uses keyword-only arguments to avoid confusion about whether position arguments are the dataset or the attributes to show:

https://github.com/glue-viz/glue-jupyter/blob/227089c802b131d1929f40137934eb5aeb45fb2a/glue_jupyter/app.py#L229

Reply:
Ah right, I forgot about that - will fix that.